frontend/src/components/user/GCUserDashboard.js

```diff
@@ -707,6 +816,73 @@ const GCUserDashboard = (props) => {
 		updateUserData();
 	}
 
+	const handleAddToGroupCheckbox = (value) => {
+		const newDocumentsToGroup = [...documentsToGroup];
+		const index = newDocumentsToGroup.indexOf(value);
+		if(index > -1){
+			newDocumentsToGroup.splice(index, 1);
+		} else {
+			newDocumentsToGroup.push(value);
+		}
+		setDocumentsToGroup(newDocumentsToGroup);
+	}
+
+	const handleDeleteGroupCheckbox = (value) => {
+		const newGroupsToDelete = [...groupsToDelete];
+		const index = newGroupsToDelete.indexOf(value);
+		if(index > -1){
+			newGroupsToDelete.splice(index, 1);
+		} else {
+			newGroupsToDelete.push(value);
+		}
+		setGroupsToDelete(newGroupsToDelete);
+	}
+
+	const handleAddToFavorites = async () => {
+		if(!selectedGroup.name) return setAddToGroupError("Please select a group");
+		const selectedGroupInfo = userData.favorite_groups.find(group => group.id === selectedGroup.id);
+		let totalInGroup = documentsToGroup.length;
+		selectedGroupInfo.favorites.forEach(favId => {
+			if(!documentsToGroup.includes(favId)) totalInGroup++;
+		})
+		if(totalInGroup > 5) return setAddToGroupError("Groups can only contain up to 5 items");
+		await gameChangerAPI.addTofavoriteGroupPOST({groupId: selectedGroup.id, documentIds: documentsToGroup});
+		updateUserData();
+		handleCloseAddGroupModal();
+	}
+
+	const handleCloseAddGroupModal = () => {
+		setAddToGroupError("");
+		setShowAddToGroupModal(false);
+		setDocumentsToGroup([]);
+	}
+
+	const renderDocumentsToAdd = () => {
+		let groupFavorites;
+		if(selectedGroup.name) groupFavorites = userData.favorite_groups.find(group => group.group_name === selectedGroup.name)?.favorites;
+		return <div style={{overflow: 'scroll', maxHeight: 300, display: 'grid', gridTemplateColumns: 'repeat(2, 1fr)'}}>
+			{_.map(favoriteDocuments, (doc) => {
+				if(selectedGroup.name){
+					if(groupFavorites.includes(doc.favorite_id)) {
```

Comment:
It only happened once, but when deleting a group I made this line errored out from groupFavorites being undefined. I think if this is handled somehow the rest looks good.

Reply:
The exporting issue is tied to a bug that I have another PR out for to fix

Reply:
Got a fix in for groupFavorites being undefined